include/openPMD/auxiliary/Visibility.hpp

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2018 Fabian Koller, Axel Huebl
+ *
+ * This file is part of openPMD-api.
+ *
+ * openPMD-api is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * openPMD-api is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with openPMD-api.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/** @{
+ * Define the compiler-specific attribute for symbol visibility.
+ *
+ * Added to the symbols of the public API.
+ * Needs to be used as qualifier for all publicly exposed, compiled
+ * functionality.
+ */
+#if defined(__GNUC__) || defined(__clang__) || defined(__ICL) || defined(__xlC__)
+#   define OPENPMDAPI_PUBLIC __attribute__((visibility("default")))
+#elif defined(_MSC_VER)
+#   define OPENPMDAPI_PUBLIC __declspec( dllexport )
+#else
+#   define OPENPMDAPI_PUBLIC
```

Comment:
maybe throw also a warning that "maybe" all symbols are exported because a program developed against the lib with all exposed symbols can be incompatible with a library version compiled with hidden symbold if non public member functions are used.

Reply:
Ok, but this would mean I will throw a warning just because e.g. PGI us used - even if only public API is properly used.